source/gui/settingsDialogs.py

```diff
@@ -1899,6 +1910,23 @@ def makeSettings(self, settingsSizer):
 		self.bindHelpEvent("ObjectPresentationReportNotifications", self.balloonCheckBox)
 		self.balloonCheckBox.SetValue(config.conf["presentation"]["reportHelpBalloons"])
 
+		# Translators: This is the label for a combo box in the
+		# object presentation settings panel.
+		uiaNotificationText = _("Report &accessible event notifications:")

```

Comment:
I don't think this is very user friendly. Now there will be the following options:
- Report notifications
- Report accessible event notifications

As a user the first question will be, how are these different? If we aren't sure, or can't find a good way to describe the difference, then I think they should be rolled together into a single setting.

Reply:
Hi,

The first option (report notifications) was originally called "report help balloons" because that's what it used to announce until Windows 7. Thanks to toasts in Windows 8.x and 10, it was renamed not long ago. The latter label came from JAWS. I think one way to reduce confusion would be renaming the first option as "Report help balloons and toast notifications" or "Report system notifications", or perhaps clarify the second option by including "UIA" i.e. "Report UIA accessible event notifications". I think renaming the first optoin, if accepted, should be part of a different issue/PR pair given the object presentation settings mess we are dealing with and add "UIA" to the second option as this setting controls UIA notification event handling.

As for the first point on categories, I can handle that in the next commit.

Thanks.

Reply:
Why not combine into a single option. I don't think users should need to have any understanding of UIA to configure whether they hear notifications. 

I guess I'm asking, is there a fundamental reason why users may want to turn on one notification technology and turn off the other? How would the user know which is which?

Reply:
Hi,

The former (report notifications) mostly concerns system notifications and pre-1709 systems where there is no UIA notification event. The latter is geared more towards app-specific notifications (even volume change notification, while delivered through UIA notification event and is system-wide, originates from File Explorer).

Also, a separate UIA notification option is desirable for notifications that are coming from an overlay window. The most prominent example is Windows Input Experience Panel in more recent Insider Preview builds (build 21296 and later to be exact; combination of emoji panel and clipboard history with additional features) where UIA notification event is fired by the overlay whenever a user changes emoji/clipboard item selection. If "report notifications" and "report accessible event notifications" are combined, by setting "report notifications" to "off" or "focused app", users will not be able to hear selected emoji/clipboard item because it is fired by an overlay that's not taking away current system focus. For this reason, in anticipation of this PR being merged, Windows 10 App Essentials add-on will simply ignore this new setting and announce "notifications" i.e. item selection from anywhere. And since app-specific config profile triggers takes foreground into account, and since the new input experience panel isn't really changing system focus/foreground, app-specific config profile setting for "report notifications" won't be effective. The ideal solution then is convincing Microsoft to utilize UIA notification event for more system-level alerts such as toasts or making Input Experience Panel handle focus changes better but that's highly unlikely for a while, as UIA notification event is really meant to be used by apps.

Thanks.

Reply:
For users who rely on notifications for emoji/clipboard, why would they have notifications turned off. Are you saying that toasts are a common problem that users must use NVDA to disable, and can't disable some other way?

Reply:
Hi,

You can't really turn off toast announcements - the shortest option is five seconds. You can in fact tell NVDA to "raise" toasts by calling wx.adv.NotificationMessage(title, message).Show() from Python Console, and toasts raise window open event. One way to "turn off" toasts is enabling Focus assist (setting it to priority only or alarms only) from Action Center.

As for emoji panel: suppose users turn off UIA notifications or chooses to set it to "focused app", particularly for Edge - turning off notifications such as "loading complete" which is delivered through UIA notification and no other way is a central tenet of the PR; rather than dealing with msedge.py app module, it was made configurable so NvDA can turn off UIA notification event handler for specific apps through app-specific config profiles defined by users. You can't really turn off UIA notifications - that's not controlled by assistive technologies unless they choose to not listen to notification event.

Thanks.

Reply:
But are excessive toast notifications a common issue for people on Windows 10 (where they want the UIA notifications)?

Reply:
> (even volume change notification, while delivered through UIA notification event and is system-wide, originates from File Explorer).

This is completely contre-intuitive from a user point of view. The user may feel that the volume setting is a system parameter whereas Windows Explorer is an application that is a file manager. Thus I think that it should be clearly documented in the user guide that global notifications such as volume, mute, airplane mode are controlled by Explorer.

Reply:
I don't think that the user should have to have any technical understanding about where notifications originate.

Reply:
Actually I think he should. A user who wants to disable volume notifications should know that he is required to create a profile for explorer.exe with this option disabled.

Reply:
I'd say that is a failing at a product design level. NVDA should be as useable and useful for all, including people who aren't developers or specifically technology oriented.

Non-tech oriented users don't perceive a UIA notification from and executable, they perceive a recent overly verbose interruption. An alternative might be to give the user a list recent interruptions and allow them to silence those kinds of interruptions in the future.